src/features/SearchResults/useSearchResults.js

```diff
@@ -0,0 +1,45 @@
+import axios from "axios";
+import { useState, useEffect } from "react";
+import { url_back, url_front_search } from "../../common/API/requests";
+import { useQueryParameter } from "../../common/Header/SearchBar/queryParameters";
+import { useLocation } from "react-router-dom/cjs/react-router-dom.min";
```

Comment:
I noticed that you are using the useLocation version from CommonJS. If possible, try using the ESM version for consistency and potential benefits in building and optimizing your application:

import { useLocation } from 'react-router-dom';

Reply:
At first when I tried importing things like useLocation, Redirect and Switch from 'react-router-dom' it was causing an error and changing it to 'react-router-dom/cjs/react-router-dom.min' solved it, but now it seems to work fine with just the react-router-dom.